src/query/select.rs

```diff
@@ -1345,3 +1366,10 @@ impl ConditionalStatement for SelectStatement {
         self
     }
 }
+
+/// List of lock types
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum LockType {
```

Comment:
Very tiny comment here. I usually put type definitions before implementations.

Reply:
I've copied the comment style for SelectDistinct

Reply:
Oh, I've realized only now you maybe intended the comments on the lock methods...
I was waiting for your approval on the design before spending time on docs and examples

Reply:
Yes, the API looks good to me.

Reply:
Ok, I've added some docs and examples